Chinese Academy of Sciences CNIC CSTR.js

```diff
@@ -0,0 +1,194 @@
+{
+	"translatorID": "c070e5a2-4bfd-44bb-9b3c-4be20c50d0d9",
+	"label": "Chinese Academy of Sciences CNIC CSTR",
+	"creator": "cheney",
+	"target": "^https?://www\\.cstr\\.cn",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 97,
+	"inRepository": true,
+	"translatorType": 12,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2022-05-11 08:00:39"
+}
+
+function detectSearch(item) {
+	return !!item.CSTR;
+}
+
+function doSearch(item) {
+	
+	let url;
+	if (item.CSTR) {
+		url = "https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=" + item.CSTR;
+	}
+	else if (item.query) {
+		url = "https://www.cstr.cn/openapi/v2/pid-common-service/statistics/metadata.search.aggregation?page=1&size=10&query=" + encodeURIComponent(item.query);
+	}
+	Z.debug(url);
+	ZU.doGet(url, parseJSON);
+}
+
+function detectWeb(doc, url) {
+	var searchRe = /^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\/(?:search\/cstrDetail\/).*/;
+	if (searchRe.test(url)) {
+		return "journalArticle";	
+	}
+	else {
+		return "multiple";
+	}
+}
+// 解析页面元素获取元数据项目
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		var rows = doc.querySelectorAll('a.identifier');
+		if(rows){
+			var urls = [];
+			for (var i = 0; i < rows.length; i++) {
+				var id = rows[i].href.replace(/^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\//,"");
+				urls.push('https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=' + encodeURIComponent(id)
+				);
+			}
+			if(urls && urls.length > 0){

```

Comment:
Reformat (`npm run lint -- "Chinese Academy of Sciences CNIC CSTR.js" --fix`) to fix formatting issues.